### ISSUE TITLE
templateParameters. Parameter is not defined

### ISSUE THREAD

**Issue by dfvgbh** [jantimon/html-webpack-plugin#915]
## Prerequisites

node: 8.10.0
webpack: 4.3.0
html-webpack-plugin: 3.1.0

## Description

Using templateParameters option gets next error:

### Error Message & Stack Trace

Html Webpack Plugin:
<pre>
  ReferenceError: foo is not defined
  
  - loader.js:4 eval
    [index.html?.]/[html-webpack-plugin]/lib/loader.js:4:11
  
  - loader.js:7 module.exports
    [index.html?.]/[html-webpack-plugin]/lib/loader.js:7:3
  
  - index.js:282 Promise.resolve.then
    [webpack]/[html-webpack-plugin]/index.js:282:18
  
  
  - next_tick.js:188 process._tickCallback
    internal/process/next_tick.js:188:7
  
</pre>

### Config

**webpack.config.js**

```
const HtmlWebPackPlugin = require('html-webpack-plugin');

module.exports = {
  plugins: [
    new HtmlWebPackPlugin({
      template: 'src/index.html',
      templateParameters: {
        foo: 'bar'
      }
    })
  ]
};
```

**src/index.html**

```
<!doctype html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport"
          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
    <meta http-equiv="X-UA-Compatible"
          content="ie=edge">
</head>
<body>
    <h1><%= foo %></h1>
</body>
</html>
```

**Comment by jantimon** [comment]
Thank you for the feedback - will look into that.  
Did you already have time to debug it a little bit?

**Comment by jantimon** [comment]
Oh thanks a lot @dfvgbh the reason is here in the hardcoded variables section:

https://github.com/jantimon/html-webpack-plugin/blob/c05b31bddc2340d7b97c5b9b5e19ba62dac82826/lib/loader.js#L35-L40

Any ideas how we could improve that?

**Comment by jantimon** [comment]
Fixed in 3.2
